akka-docs/src/test/java-jdk9-only/jdocs/stream/operators/source/FromPublisher.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.stream.operators.source;
+
+//#imports
+import java.util.concurrent.Flow.Publisher;
+
+import akka.NotUsed;
+import akka.stream.javadsl.Source;
+import akka.stream.javadsl.JavaFlowSupport;
+
+//#imports
+
+import org.apache.commons.lang.NotImplementedException;
+
+public interface FromPublisher {
+    static class JavaFlowSupport {
+        public static final class Source {
+            public
+            // #api
+            static <T> akka.stream.javadsl.Source<T, NotUsed> fromPublisher(Publisher<T> publisher)
+            // #api
+            {
+                return akka.stream.javadsl.JavaFlowSupport.Source.<T>fromPublisher(publisher);
+            }
+        }
+    }
+
+    static class Row {
+        public String getField(String fieldName) {
+            throw new NotImplementedException();
+        }
+    }
+
+    static class DatabaseClient {
+        Publisher<Row> fetchRows() {
+            throw new NotImplementedException();
+        }
+    }
+
+    DatabaseClient databaseClient = null;
+
+    // #example
+    class Example {
+        public Source<String, NotUsed> names() {
+            // rowSource can be re-used, since it will start a new
+            // query for each materialization, fully supporting backpressure
+            // for each materialized stream:
```

Comment:
AFAICS there is no guarantee the client db API would work like that, it could create a single publisher that can't really be re-used like this and would be consumed after the first stream has been materialized. So maybe state that more like an aspect of our pretend db?

(BTW `Source.lazySource(() -> Source.fromPublisher(databaseClient.fetchRows()))` could solve that but not a great thing to push in a sample perhaps)